scipy/optimize/optimize.py

```diff
@@ -2397,20 +2399,76 @@ def bracket(func, xa=0.0, xb=1.0, args=(), grow_limit=110.0, maxiter=1000):
     return xa, xb, xc, fa, fb, fc, funcalls
 
 
-def _linesearch_powell(func, p, xi, tol=1e-3):
+def _line_for_search(x0, alpha, lower_bound, upper_bound):
+    """
+    Given an n-dimensional parameter vector x0 and an n-dimensional
+    direction vector alpha, and lower and upper bounds on each of
+    the n parameters, what are the bounds on a scalar l such that
+    lower_bound <= x0 + alpha * l <= upper_bound.
+
+    x0 is the vector representing the current location (np.array)
+    alpha is the vector representing the direction (np.array)
+
+    ie the direction is along the line from x0 to alpha
+
+    lower_bound is a np.array of the lower bounds for each parameter in x0.
+    upper_bound is a np.array of the upper bounds for each parameter in x0.
+
+    returns (lmin, lmax), the bounds for l such that
+            lower_bound[i] <= x0_i+alpha_i * l <= upper_bound[i]
+        for all i.
+    """
+    # get nonzero indices of alpha so we don't get any zero division errors.
+    # alpha will not be all zero, since it is called from _linesearch_powell
+    # where we have a check for this.
+    nonzero, = alpha.nonzero()
+    lower_bound, upper_bound = lower_bound[nonzero], upper_bound[nonzero]
+    x0, alpha = x0[nonzero], alpha[nonzero]
+    low = (lower_bound - x0) / alpha
+    high = (upper_bound - x0) / alpha
+
+    # positive and negative indicies
+    pos, neg = alpha > 0, alpha < 0
+    lmin = np.max(low * pos + high * neg)
+    lmax = np.min(low * neg + high * pos)
+
+    return lmin, lmax
+
+
+def _linesearch_powell(func, p, xi, fval,
```

Comment:
`_linesearch_powell` doesn't have needs to have a test suite in `test_optimize`, could this PR add one?

Reply:
Done.